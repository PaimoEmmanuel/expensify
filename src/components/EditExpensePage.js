import React from 'react';
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses'
import { startRemoveExpense } from '../actions/expenses';


export class EditExpensePage extends React.Component {

  onSubmit = (expense) => {
    this.props.editExpense(this.props.match.params.id, expense);
    this.props.history.push('/');
  }
  onRemove = () => {
      this.props.startRemoveExpense({ id: this.props.match.params.id });
      this.props.history.push('/');
  }
  render() {
    return (
      <div>
      Editing the expense with id of {this.props.match.params.id}
      <ExpenseForm 
      expense={this.props.expense}
      onSubmit = {this.onSubmit}/>
      <button onClick={this.onRemove}>Remove</button>
    </div>
    )
  }
}


const mapStateToProps = (state, props) =>{
  return {
    expense: state.expenses.find((expense => expense.id === props.match.params.id))
  }
}
const matchDispatchToProps = (dispatch, props) => {
  return {
    editExpense: (id, updates) => dispatch(editExpense(id, updates)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
  }
}
export default connect(mapStateToProps, matchDispatchToProps)(EditExpensePage);
