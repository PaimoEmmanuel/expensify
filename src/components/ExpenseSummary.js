import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = (props) => {
    console.log(props.expenseTotal);
    const expenseWord = props.expenseCount > 1 ? 'expenses' : 'expense';
     const formattedExpenseTotal = numeral(props.expenseTotal / 100).format('$0,0.00');
    return (

        <div>
            <p>Viewing {props.expenseCount} {expenseWord} totalling {formattedExpenseTotal}</p>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return{
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpensesTotal(visibleExpenses) 
    }
};
export default connect(mapStateToProps)(ExpenseSummary);
