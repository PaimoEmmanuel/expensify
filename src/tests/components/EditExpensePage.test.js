import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import { expenses } from '../fixtures/expenses'

let expense, startEditExpense, startRemoveExpense, match, history, wrapper;
beforeEach(() => {
    expense = expenses[1];
    startEditExpense = jest.fn();
    startRemoveExpense = jest.fn();
    match = {
        params: {
            id: expenses[1].id
        }
    };
    history = { push: jest.fn() }
    wrapper = shallow(<EditExpensePage 
        expense = {expense} 
        startEditExpense ={startEditExpense} 
        startRemoveExpense= {startRemoveExpense} 
        match = {match}
        history = {history}/>);
});
test('Should render EditExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should handle startEditExpense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expense);
    expect(startEditExpense).toHaveBeenLastCalledWith(expense.id, expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
});

test('Should handle startRemoveExpense', () => {
    wrapper.find('button').prop('onClick')();
    expect(startRemoveExpense).toHaveBeenLastCalledWith({ id: expense.id });
    expect(history.push).toHaveBeenLastCalledWith('/');
})