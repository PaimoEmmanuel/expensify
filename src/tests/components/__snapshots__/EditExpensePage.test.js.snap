// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render EditExpensePage correctly 1`] = `
<div>
  Editing the expense with id of 
  2
  <ExpenseForm
    expense={
      Object {
        "amount": 1500,
        "createdAt": 0,
        "description": "Paid my last rent",
        "id": "2",
        "text": "Rent",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
